### DISCUSSION TITLE
Unable to access value in rule type validate

### DISCUSSION THREAD

**Discussion by kashif-sk** [react-hook-form/react-hook-form#8777] [General] [answered]
I am updating my value which depends on previous value in this manner `onPress={() => onChange((prevVal) => prevVal + 1)}`

This updates the values perfectly
But while setting rules, rule type validate receives the function as parameter instead of value
eg:
```
validate: (value) => {
            console.log(value);
            return value === 1 || 'Error';
},
```

Here the expected output should be the entered value but it prints the whole function

Actual output:
```
ƒ (prevVal) {
              return prevVal + 1;
 }
```

Am I missing something ?

react-hook-form v7.34.0
using with react native

**Comment by bluebill1049** [comment, marked as answer]
use `getValues()` function its very cheap to run

**Reply by Moshyfawn** [reply]
To expand a bit, the field's `onChange` handler doesn't use the "prev value callback syntax" like you tried to use `(prevValue) => prevValue + 1`; it's not `setState`, it's a basic setter: you can pass a value as it's argument and that's it

**Reply by kashif-sk** [reply]
@bluebill1049 @Moshyfawn
Got it. Thanks!!
